pandas/tseries/tests/test_base.py

```diff
@@ -617,6 +617,23 @@ def test_nat_new(self):
         exp = np.array([tslib.iNaT] * 5, dtype=np.int64)
         tm.assert_numpy_array_equal(result, exp)
 
+    def test_shift(self):
+        # GH 9903
+        for tz in [None, 'US/Eastern', 'Asia/Tokyo']:
+            idx = pd.DatetimeIndex([], name='xxx', tz=tz)
+            tm.assert_index_equal(idx.shift(0, freq='H'), idx)
```

Comment:
I think you want `exact=True` in all the `assert_index_equal` calls here.  By default, `assert_index_equal` only asserts that both values are instances of `Index`.